### PR TITLE
released groovy file

### PR DESCRIPTION
**Pull request by tintumathew10**
Signed-off-by: tintumathew10 <tmathew@redhat.com>

Rhceph released groovy script.

logs:
https://ceph-downstream-jenkins-csb-storage.apps.ocp4.prod.psi.redhat.com/view/RHCS%205%20QE/job/rhcs-tintu/6/console
https://datagrepper.engineering.redhat.com/id?id=ID:ceph-downstream-jenkins-csb-storage-35005-1631702602504-3895:1:1:1:1&is_raw=true&size=extra-large

### CODE REVIEW

File: pipeline/released.groovy [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// Script to update the container build information into QE recipes.
```
Comment (psathyan) [inline pipeline/released.groovy:1]: Could we fix the documentation? I believe the purpose of this script is to trigger when a RH Ceph is released and execute Tier-0 using the bits available in the external repository.

File: pipeline/released.groovy [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                userRemoteConfigs: [[
+                    url: 'https://github.com/red-hat-storage/cephci.git'
+                ]]
+            ])
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+            sharedLib.prepareNode()
+        }
+    }
+
+    stage("Prepare Tier-0 suite") {
```
Comment (psathyan) [inline pipeline/released.groovy:42]: IMO, the stage name could be `Get test suites`

File: pipeline/vars/lib.groovy [hunk 3]
```diff
@@ -333,12 +338,20 @@ def fetchStages(def scriptArg, def tierLevel, def testResults) {
...
+    def minorVersion
+    if (scriptArg == 'released') {
+        def versions = fetchMajorMinorOSVersion("released")
+        majorVersion = versions.major_version
+        minorVersion = versions.minor_version
+    } else {
+        def RHCSVersion = getRHCSVersionFromArtifactsNvr()
+        majorVersion = RHCSVersion["major_version"]
+        minorVersion = RHCSVersion["minor_version"]
+    }
```
Comment (psathyan) [inline pipeline/vars/lib.groovy:351]: We could rewrite this like
```
def RHCSVersion = [:]

if (scriptArg == "released") {
  RHCSVersion = getRHCSVersionFromArtifactsNvr()
} else {
  RHCSVersion = fetchMajorMinorOSVersion("released")
}
def majorVersion = RHCSVersion.major_version
def minorVersion = RHCSVersion.minor_version
```

File: pipeline/vars/lib.groovy [hunk 4]
```diff
@@ -258,17 +271,9 @@ def sendEmail(def testResults, def artifactDetails, def tierLevel){
...
-    body += "<table>"
-    body += "<tr><th>Test Suite</th><th>Result</th></tr>"
-    for (test in testResults) {
-        def test_name = test.key.replace("-", " ")
-        body += "<tr><td>${test_name.capitalize()}</td><td>${test.value}</td></tr>"
-    }
     body += "</table><br /></body></html>"
     if ('FAIL' in testResults.values()){
-        toList = "ceph-qe@redhat.com"
+        toList = "tmathew@redhat.com"
```
Comment (psathyan) [inline pipeline/vars/lib.groovy:276]: Could you please undo this change?

File: pipeline/released.groovy [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            ],
+            "test": [
+                "category": "release",
+                "result": status
+            ],
+            "version": "1.1.0"
+        ]
+
+        def msgContent = writeJSON returnText: true, json: contentMap
+        def overrideTopic = "VirtualTopic.qe.ci.rhcephqe.test.complete"
```
Comment (psathyan) [inline pipeline/released.groovy:97]: IMO, let's publish it to `VirtualTopic.qe.ci.rhceph.test.complete`

**Review by psathyan — COMMENTED** [review]
Looks good to me... some minor changes

File: pipeline/vars/lib.groovy (hunk 4 above)
Comment (tintumathew10) [reply pipeline/vars/lib.groovy]: done

**Review by psathyan — APPROVED** [review]
